File: Leetcode/temp/0448_Find_All_Numbers_Disappeared_in_an_Array.py
class Solution(object):
    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        if not nums:
            return []
        return list(set(list(range(1, len(nums) + 1))) - set(nums))

File: Leetcode/Java/0448_Find_All_Numbers_Disappeared_in_an_Array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> miss = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            int index = Math.abs(nums[i]) - 1;
            if (nums[index] > 0) {
                nums[index] = -nums[index];
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > 0) {
                miss.add(i + 1);
            }
        }
        return miss;
    }
}